Ghidra/Processors/CP1600/data/languages/CP1600.slaspec

```diff
@@ -0,0 +1,651 @@
+	local tmpS = (reg0_2 & 0x4000) != 0;
+	reg0_2 = reg0_2 << 1;
+	C = tmpC;
+	S = tmpS;
+	Z = reg0_2 == 0;
+} 
+
+:SLR reg0_2, 1 is opcode3_9=0x000C & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local tmpS = (reg0_2 & 0x0100) != 0;
+	reg0_2 = reg0_2 >> 1;
+	S = tmpS;
+	Z = reg0_2 == 0;
+} 
+
+:SWAP reg0_2, 1 is opcode3_9=0x0008 & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local tmpS = (reg0_2 & 0x8000) != 0;
+	local tmp = (reg0_2 << 8) & 0xFF00;
+	reg0_2 = tmp | ((reg0_2 >> 8) & 0x00FF);
+	S = tmpS;
+}
+
+}
+
+with : operation_size=1 {
+
+:RLC reg0_2, 2 is opcode3_9=0x000A & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local tmpC = (reg0_2 & 0x8000) != 0;
+	local tmpO = (reg0_2 & 0x4000) != 0;
+	local tmpS = (reg0_2 & 0x2000) != 0;
+	reg0_2 = (reg0_2 << 2) + (zext(C) << 1) + zext(O);
+	C = tmpC;
+	O = tmpO;
+	S = tmpS;
+	Z = reg0_2 == 0;
+}
+
+:RRC reg0_2, 2 is opcode3_9=0x000E & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local tmpC = (reg0_2 & 0x0001) != 0;
+	local tmpO = (reg0_2 & 0x0002) != 0;
+	local tmpS = (reg0_2 & 0x0200) != 0;
+	reg0_2 = (reg0_2 >> 2) | (zext(C) << 14) | (zext(O) << 15);
+	C = tmpC;
+	O = tmpO;
+	S = tmpS;
+	Z = reg0_2 == 0;
+}
+
+:SAR reg0_2, 2 is opcode3_9=0x000D & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local tmpS = (reg0_2 & 0x0200) != 0;
+	reg0_2 = reg0_2 s>> 2;
+	S = tmpS;
+	Z = reg0_2 == 0;	
+}
+
+:SARC reg0_2, 2 is opcode3_9=0x000F & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local tmpC = (reg0_2 & 0x0001) != 0;
+	local tmpO = (reg0_2 & 0x0002) != 0;
+	local tmpS = (reg0_2 & 0x0200) != 0;
+	reg0_2 = reg0_2 s>> 2;
+	C = tmpC;
+	O = tmpO;
+	S = tmpS;
+	Z = reg0_2 == 0;
+}
+
+:SLL reg0_2, 2 is opcode3_9=0x0009 & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local tmpS = (reg0_2 & 0x2000) != 0;
+	reg0_2 = reg0_2 << 2;
+	S = tmpS;
+	Z = reg0_2 == 0;
+}
+
+:SLLC reg0_2, 2 is opcode3_9=0x000B & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local tmpC = (reg0_2 & 0x8000) != 0;
+	local tmpO = (reg0_2 & 0x4000) != 0;
+	local tmpS = (reg0_2 & 0x2000) != 0;
+	reg0_2 = reg0_2 << 2;
+	C = tmpC;
+	O = tmpO;
+	S = tmpS;
+	Z = reg0_2 == 0;
+} 
+
+:SLR reg0_2, 2 is opcode3_9=0x000C & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local tmpS = (reg0_2 & 0x0200) != 0;
+	reg0_2 = reg0_2 >> 2;
+	S = tmpS;
+	Z = reg0_2 == 0;
+}
+
+:SWAP reg0_2, 2 is opcode3_9=0x0008 & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local tmpS = (reg0_2 & 0x0080) != 0;
+	reg0_2 = (reg0_2 << 8) | (reg0_2 & 0x00FF);
+	S = tmpS;
+}
+
+}
+
+with : doublebyte=0 {
+
+:ADD@ reg3_5, reg0_2 is opcode6_9=0x000B & reg0_2 & reg3_5 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local val:2 = 0;
+	readHalfDoubleWord(reg3_5, val);
+	addition(reg0_2, val);
+}
+
+:ADDI #imm16, reg0_2 is opcode3_9=0x005F & reg0_2 ; imm16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
```

Comment:
The "#" before imm16 is unnecessary here (and in the following instructions). Ghidra will automatically insert it before the immediate where appropriate.

Reply:
Well, I wasn't actually able to find a case where Ghidra would put that prefix itself.  I also noticed that other processor modules explicitly put that (although it's enclosed by quotes) in the operand definition table entries and yet said prefix shows in listings for those processors (6502 has the same set of rules for literals as CP1600, for example).